### PR TITLE
fix: allow empty structs in schema inference

### PR DESCRIPTION
**Pull request by agavra**
### Description 
Previously, empty structs would be formatted as `STRUCT<>` in the `SqlSchemaFormatter` (which for some reason is used independently of `SqlFormatter`, we should dig into that) which would cause parsing to fail because `<>` is it's own symbol for "not equal to".

This code path was not used in master after https://github.com/confluentinc/ksql/pull/4637 so it was no longer happening. That change has a relatively large surface area, so for the backport I am making the most targeted change possible to address the edge case.

### Testing done 
Added a unit test, which fails before this fix.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java [hunk 1]
```diff
@@ -161,6 +161,9 @@ public String visitMap(final Schema schema, final String key, final String value
     }
 
     public String visitStruct(final Schema schema, final List<? extends String> fields) {
+      if (fields.isEmpty()) {
+        return "STRUCT< >";
+      }
```
Comment (AlanConfluent) [inline ksqldb-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java:166]: Just curious, what does the below collector return with empty fields?

Comment (agavra) [reply ksqldb-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java:166]: `STRUCT<>`
